x-pack/plugins/fleet/common/authz.test.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Capabilities } from '@kbn/core-capabilities-common';
+
+import {
+  calculatePackagePrivilegesFromCapabilities,
+  calculatePackagePrivilegesFromKibanaPrivileges,
+} from './authz';
+import { ENDPOINT_PRIVILEGES, SECURITY_SOLUTION_ID } from './constants';
+
+function generateActions(privileges: string[] = [], overrides: Record<string, boolean> = {}) {
+  return privileges.reduce((acc, privilege) => {
+    const executePackageAction = overrides[privilege] || false;
+
+    return {
+      ...acc,
+      [privilege]: {
+        executePackageAction,
+      },
+    };
+  }, {});
+}
+
+describe('fleet authz', () => {
+  describe('calculatePackagePrivilegesFromCapabilities', () => {
+    it('calculates privileges correctly', () => {
+      const endpointCapablities = {
+        writeEndpointList: true,
+        writeTrustedApplications: true,
+        writePolicyManagement: false,
+        readPolicyManagement: true,
+        writeHostIsolationExceptions: true,
+        writeHostIsolation: false,
+      };
+      const expected = {
+        endpoint: {
+          actions: generateActions(ENDPOINT_PRIVILEGES, endpointCapablities),
+        },
+      };
+      const actual = calculatePackagePrivilegesFromCapabilities({
+        siem: endpointCapablities,
+      } as unknown as Capabilities);
```

Comment:
Could you create a fake capabilities object instead of doing this cast? Something like:

```typescript
const actual = calculatePackagePrivilegesFromCapabilities({
        navLinks: { },
        management: { },
        catalogue: { },
        siem: endpointCapablities,
      });
```

Reply:
sure 👍